src/query-builder/InsertQueryBuilder.ts

```diff
@@ -360,7 +373,8 @@ export class InsertQueryBuilder<Entity> extends QueryBuilder<Entity> {
                 && !(this.connection.driver instanceof OracleDriver)
                 && !(this.connection.driver instanceof AbstractSqliteDriver)
                 && !(this.connection.driver instanceof MysqlDriver)
-                && !(this.connection.driver instanceof AuroraDataApiDriver))
+                && !(this.connection.driver instanceof AuroraDataApiDriver)
+                && !(this.connection.driver instanceof SqlServerDriver && this.getValueSets().some((valueSet) => column.getEntityValue(valueSet) !== undefined && column.getEntityValue(valueSet) !== null)))
```

Comment:
Yes it is. There are three scenarios:

- You have a column which is not an entity value, then including the column is fine.

- You have a column which is an entity value, but you are not specifying a valid value for that (i.e. `undefined` or `null`). Then the primary key must be generated automatically. To get this working in mssql, you are not allowed to include the column name in the query, otherwise it will complain because you are not allowed to insert something unless `IDENTITY_INSERT` is enabled. However if you enable that, it will not generate a value anymore.

- You have a column which is an entity value, and you are specifying a value for that. Then `IDENTITY_INSERT` will be enabled and you must specify the column name.

I hope that helps. If not, please let me know. I spent some time on figuring out the behavior of mssql server with regard to that `IDENTITY_INSERT` option.